server/src/main/java/org/elasticsearch/common/io/stream/ReleasableBytesStreamOutput.java

```diff
@@ -37,4 +37,10 @@ public ReleasableBytesStreamOutput(int expectedSize, BigArrays bigArrays) {
     public void close() {
         Releasables.close(bytes);
     }
+
+    @Override
+    public void reset() {
+        // not supported, close and create a new instance instead
+        throw new UnsupportedOperationException("must not reuse a pooled bytes backed stream");
```

Comment:
To me it doesn't look like down-sizing a pooled `BigArray` or so works correctly. This is the safest option I could find for the time being.

Reply:
++